tests/ui/consts/custom-suffixes-exponent-like.rs

```diff
@@ -0,0 +1,10 @@
+const _A: f64 = 1em;
+    //~^ ERROR invalid suffix `em` for number literal
+const _B: f64 = 1e0m;
+    //~^ ERROR invalid suffix `m` for float literal
+const _C: f64 = 1e_______________0m;
+    //~^ ERROR invalid suffix `m` for float literal
+const _D: f64 = 1e_______________m;
+    //~^ ERROR invalid suffix `e_______________m` for number literal
```

Comment:
There should be a test checking that all these newly accepted suffixes can be consumed by macros, like `do_nothing!(1e 1e2e4 1eow)` ,  without emitting errors.